wordpress-mobile/WordPress-Android
# Unified Login & Sign-Up: Login library theme and styles update

Fixes #11785

Depends on #11944

This PR updates the Login Library's theme and various component styles, and also bumps its `minSdkVersion` to 21 and `targetSdkVersion` to 29, since those are already the versions used by both of its clients apps (WPAndroid and WCAndroid).

The theme updates make use of the new Material Design system for colors, typography and shapes, and is now better separated, which should make a lot easier to extend it and override specific theme attributes and styles from now on. It also includes a dark theme.

Both the theme and the styles were renamed to reflect best practices regarding theming on Android, so most of the changes on layout files are due to those changes.

The key changes are in the following files under the `res/values` folder of the login library:
- styles.xml
- themes.xml
- types.xml

Old login theme overloads were removed from the app, since the new theme was based on the look we currently want to achieve.

## To test

These changes are mostly on the code side rather than on the UI, but you can simply navigate through the login flow and make sure its appearance is mostly the same as it was here: #11944.

## Notes
- This PR does not include any changes do the login flow screens layouts. Those are being handled separatedly here: #12048
- Styles that were being used only by the prologue and the epilogue screen were removed from the library, since it doesn't include those screens.

---

PR submission checklist:
